pairinteraction_gui/pairinteraction/app.py

```diff
@@ -1081,1226 +1111,1200 @@ def getConstDistance(self):
-                    if idx == 0 or idx == 1:
-                        # calculate which momenta appear in a basis element
-                        momentum_probabilty = probs.T * self.momentummat[idx]  # nBasis, nMomentum
-
-                        # keep information about the momentum that appears with
-                        # a probability > 0.5 only
-                        momentum_probabilty.data[:] *= momentum_probabilty.data > 0.5
-                        momentum_probabilty.eliminate_zeros()
-                        momentum_probabilty = sparse.coo_matrix(momentum_probabilty)
-
-                        # store the value of this momentum
-                        # -1 means no determinable momentum
-                        momentum = -np.ones(momentum_probabilty.shape[0], dtype=int)
-                        momentum[momentum_probabilty.row] = momentum_probabilty.col
-
-                    # --- calculate the position (x value) ---
-                    if self.xAxis[idx] in ["B", "E"]:
-                        rotator = np.array(
-                            [
-                                [np.cos(-self.angle), 0, -np.sin(-self.angle)],
-                                [0, 1, 0],
-                                [np.sin(-self.angle), 0, np.cos(-self.angle)],
-                            ]
-                        )
+            if self.xAxis[idx] == "B":
+                fields = [
+                    [float(params.get("Bx", params.get("atom1.Bx")))],
+                    [float(params.get("By", params.get("atom1.By")))],
+                    [float(params.get("Bz", params.get("atom1.Bz")))],
+                ]
+                fields = np.dot(rotator, fields).flatten()
+                position = self.get1DPosition(fields) * self.converter_x[idx]
+            elif self.xAxis[idx] == "E":
+                fields = [
+                    [float(params.get("Ex", params.get("atom1.Ex")))],
+                    [float(params.get("Ey", params.get("atom1.Ey")))],
+                    [float(params.get("Ez", params.get("atom1.Ez")))],
+                ]
+                fields = np.dot(rotator, fields).flatten()
+                position = self.get1DPosition(fields) * self.converter_x[idx]
+            elif self.xAxis[idx] == "R":
+                position = float(params.get("R", params.get("pair.distance"))) * self.converter_x[idx]
+
+            # --- draw labels at the beginning of the plotting ---
+            if self.ui.groupbox_plot_labels.isChecked() and filestep == 0:
+
+                # probability to find a label inside a basis element
+                if not hasattr(self, "labelprob_energy") or self.labelprob_energy is None:
+                    # nBasis, nLabels # TODO !!! tocsr
+                    self.labelprob = (probs.T * self.labelmat[idx]).tocsr()
+                    self.labelprob_energy = [energies]
+                else:
+                    csr_vappend(self.labelprob, (probs.T * self.labelmat[idx]).tocsr())
```

Comment:
If different symmetry sectors have a different number of basis elements, the dimension of `self.labelstates` and thus `self.labelmat` are different for the different symmetry sectors. Then, `csr_vappend` produces an invalid `self.labelprob` object that leads to a segfault lateron in the code.

Reply:
fixed by setting labelprob,... as dict with one entry per symmetry block, now plotting labels for each symmetry block separately